frontend/src/components/button-group/button-group.tsx

```diff
@@ -24,12 +24,14 @@ export function ButtonGroup({
       }}
     >
       {React.Children.map(children, child => {
-        return React.cloneElement(child as any, {
-          style: {
-            flexGrow: 1,
-            flexBasis: 0
-          }
-        })
+        if (child) {
```

Comment:
I don't think this check is necessary. React.children.map will only call the function on each child if children is an array, and would already have returned a null of undefined if not. Perhaps checking that children has a length greater than 0 would suffice? i.e. 
`{ children.length > 0 && React.Children.map(children, child => {`

Reply:
Yes indeed. And I don't think we event need to verify `children.length > 0` . The `map()` function is not supposed to trigger the closure if the collection is empty.

Reply:
That's right, the condition the map function will be attempted and return a null or undefined, that would be fine too.

Reply:
Hm so this was because I was getting an error `Cannot clone element of null`, as one of the children is conditionally rendered.. I will look into it again, perhaps theres a better way

Reply:
So I believe we absolutely need an if statement here.

Heres the type definition for `child` provided by `React.Children.map`

`(parameter) child: string | number | boolean | {} | React.ReactElement<any, string | React.JSXElementConstructor<any>> | React.ReactPortal | null | undefined`

null and undefined are valid react children but _not_ valid arguments for React.cloneElement. I'd like the ButtonGroup component be able to accept null or undefined children so that we can optionally render some children within ButtonGroup ([see here](https://github.com/vegaprotocol/vegawallet-desktop/blob/40f0bca90037412a228b6630ade98553c7bdc486/frontend/src/components/onboard/onboard.tsx#L92)). 

In those cases if the child is null we simply want to omit it.

To make this clearer if changed from checking truthy-ness of child and instead used React.isValidElement.